File: JavaScript/7js 1/cartesianNeighbor.js
/*
|--------------------------------------------------------------------------
| Cartesian neighbors
|--------------------------------------------------------------------------
|
*/

function cartesianNeighbor(x, y) {
   return [
      [x-1, y+1], [x, y+1], [x+1, y+1],
      [x-1, y], [x+1, y],
      [x-1, y-1], [x, y-1], [x+1, y-1],
   ];
}


function cartesianNeighbor(x, y) {
   let arr = [];
   for (let i = x - 1; i <= x + 1; i++) {
      for (let j = y - 1; j <= y + 1; j++) {
         arr.push([i, j]);
      }
   }
   return arr.filter((f) => f + "" != [x, y] + "");
}


function cartesianNeighbor(x, y) {
   return [-1, 0, 1]
      .reduce(
         (s, v) => s.concat([-1, 0, 1].reduce((si, vi) => si.concat([[x + v, y + vi]]), [])),
         []
      )
      .filter((v) => v[0] !== x || v[1] != y);
}


console.log(cartesianNeighbor(2, 2), [
   [1, 1], [1, 2], [1, 3], [2, 1], [2, 3], [3, 1], [3, 2], [3, 3],
]);

File: Python/7py 1/cartesian_neighbor.py
"""
|--------------------------------------------------------------------------
| Cartesian neighbors
|--------------------------------------------------------------------------
|
"""

def cartesian_neighbor(x,y):
   return [
      (x-1, y+1), (x, y+1), (x+1, y+1),
      (x-1, y), (x+1, y),
      (x-1, y-1), (x, y-1), (x+1, y-1),
   ]


def cartesian_neighbor(x,y):
   return [(x+i, y+j) for i in range(-1,2) for j in range(-1,2) if i or j]


def cartesian_neighbor(x,y):
   arr = []
   for i in range(x-1, x+2):
      for j in range(y-1, y+2):
         if (i,j) != (x,y): arr.append((i,j))
   return(arr)


print(cartesian_neighbor(2, 2), [
   [1, 1], [1, 2], [1, 3], [2, 1], [2, 3], [3, 1], [3, 2], [3, 3],
])